tqp/topic_queue_poller.py

```diff
@@ -214,11 +266,21 @@ def decorator(func):
 
         return decorator
 
+    def s3_handler(self, bucket_name):
+        def decorator(func):
+            if bucket_name in self.s3_handlers:
```

Comment:
Should/could this check be outside the decorator?

Reply:
no because this would allive to call the decorator twice:

```py
make_handler = poller.s3_handler('foo')
make_handler(f1)
make_handler(f2)
```

Reply:
Oh right, the decorator is evaluated on module import. I thought it was evaluated at function invocation so was worried, this would fail at "run time"